src/MergeContext.js

```diff
@@ -973,8 +973,18 @@ class PullRequest {
                 statusChecks.addOptionalStatus(new StatusCheck(st));
         }
 
+        // Returns whole check runs history for the commit.
+        // We need only the freshest checks.
         const checkRuns = await GH.getCheckRuns(this._prHeadSha());
-        for (let st of checkRuns) {
+        // check.id is greater in newer checks
+        const sortedCheckRuns = checkRuns.sort((e1, e2) => parseInt(e1.id) - parseInt(e2.id));
+        let uniqueCheckRuns = [];
+        for (let i = sortedCheckRuns.length -1; i >= 0; --i) {
```

Comment:
I hope this explicit loop will go away with other changes, but if it does not, please add a space:

```suggestion
        for (let i = sortedCheckRuns.length - 1; i >= 0; --i) {
```